### ISSUE TITLE
side_effect Does Not Work With autospec

### ISSUE THREAD

**Issue by GoogleCodeExporter** [testing-cabal/mock#242]
```
What steps will reproduce the problem?
1. Run the attached file: python test.py

What is the expected output? What do you see instead?
Expected: Both tests pass.
Result: One test fails with this stack trace:

===================================================
ERROR: test_show_side_effect_bug (__main__.BugTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 18, in test_show_side_effect_bug
    self.assertIs(A().f1(), 1)
  File "<string>", line 3, in f1
  File "python2.7/site-packages/mock.py", line 955, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "python2.7/site-packages/mock.py", line 1013, in _mock_call
    result = next(effect)
TypeError: list object is not an iterator

----------------------------------------------------------------------

What version of the product are you using? On what operating system?
Mock==1.0.1
OSX 10.9.5

Please provide any additional information below.
Doesn't seem to be a high priority, since the workaround is easy. I'm guessing 
fix will come either around line 1007 in mock.py, or somewhere earlier in the 
autospec-specific code.
```

Original issue reported on code.google.com by `kenny.tr...@workiva.com` on 22 Dec 2014 at 7:08

Attachments:
* [test.py](https://storage.googleapis.com/google-code-attachments/mock/issue-242/comment-0/test.py)

**Comment by rbtcollins** [comment]
#264 may be a dupe of this.

**Comment by rbtcollins** [comment]
Right, so this is the crux of the 1.1 issue I think. There's no test in the stdlib showing this, so I'm going to add the test there to lock the API in and then backport it + figure a fix.

**Comment by rbtcollins** [comment]
The test script passes for me with Python 2.7.6 and 2.7.8 and mock 1.1.2. So instead, I'm going to close this as fixed and focus directly on #264.
